Implement dashboard features and refactor recipe handling

gurkirtheerey opened the pull request:
- Added a new DashboardContent component to manage the dashboard layout and functionality.
- Introduced CreateRecipeModal for creating new recipes with form handling and validation.
- Refactored the recipe creation logic by removing the old createRecipe function from actions.ts.
- Updated package.json and package-lock.json to include @radix-ui/react-label and other dependency updates.
- Created reusable UI components for dialog and label to enhance the user interface.

gurkirtheerey committed:
Implement dashboard features and refactor recipe handling

- Added a new DashboardContent component to manage the dashboard layout and functionality.
- Introduced CreateRecipeModal for creating new recipes with form handling and validation.
- Refactored the recipe creation logic by removing the old createRecipe function from actions.ts.
- Updated package.json and package-lock.json to include @radix-ui/react-label and other dependency updates.
- Created reusable UI components for dialog and label to enhance the user interface.

gurkirtheerey committed:
Remove EmailConfirmationPage component and related logic from the welcome page

sunshinemoua commented on app/dashboard/DashboardContent.tsx:
```diff
@@ -0,0 +1,23 @@
+      <div className="flex justify-between p-4 w-full">
+        <h1 className="text-2xl font-bold">Dashboard</h1>
+        <div className="flex justify-center gap-4">
+          <Button>Discover</Button>
```
Route to /discover after my PR is merged

sunshinemoua commented on components/Recipe/CreateRecipeModal.tsx:
```diff
@@ -0,0 +1,90 @@
+              <Label htmlFor="instructions">Instructions</Label>
+              <Textarea id="instructions" name="instructions" placeholder="Recipe instructions" required />
+            </div>
+          </div>
```
Add image upload in this modal. Can come back to this later

Also, add input for cook time, prep time, total time, servings number

sunshinemoua commented on hooks/useAuth.ts:
add comments

sunshinemoua reviewed (changes requested):
Just a few little things to update. Otherwise, good job!

gurkirtheerey committed:
Add Sonner for toast notifications and update dashboard links
- Integrated the Sonner library for toast notifications in CreateRecipeModal.
- Updated DashboardContent to include a link for the Discover button.
- Added Sonner to package.json and package-lock.json for dependency management.
- Enhanced layout.tsx to include the Toaster component for displaying notifications.

gurkirtheerey committed:
Enhance recipe creation with form validation and new dependencies

- Integrated react-hook-form and zod for form handling and validation in CreateRecipeModal.
- Added createRecipeSchema for structured recipe data validation.
- Updated package.json and package-lock.json to include @hookform/resolvers and zod for dependency management.
- Refactored form submission logic to utilize the new validation schema.
